src/lib/utils.js

```diff
@@ -0,0 +1,71 @@
+import rc from 'rc';
+import fs from 'fs';
+import gitRemoteOriginUrl from 'git-remote-origin-url';
+import getUrlFromGitRemote from './git';
+import loadPropertiesFile from './properties-file';
+
+const gitlabEnvFileName = 'gitlab.env.yml';
+
+async function getConf(cmd, bSingle = false) {
+  const conf = rc('gitlab');
+  const errors = [];
+
+  // Check for token
+  if (!conf.token) {
+    if (!process.env.GITLAB_TOKEN) {
+      errors.push('No Gitlab token given.');
+    } else {
+      conf.token = process.env.GITLAB_TOKEN;
+      console.log('Using token from environment variable GITLAB_TOKEN.');
+    }
+  }
+
+  if (conf.config) {
+    console.log(`Using token from ${conf.config}.`);
+  }
+
+  // If there is no url provided, get it!
+  if (!conf.url) {
+    try {
+      conf.url = await getUrlFromGitRemote(gitRemoteOriginUrl);
+      console.log('No URL specified, using git remote `origin`.');
+    } catch (err) {
+      errors.push('No Gitlab project URL given.');
+    }
+  }
+
+  if (bSingle) {
```

Comment:
This logic does not belong here. A property (i.e. `key/value` combination) is not part of config (i.e. information on how to connect to gitlab and how to handle existing values). This logic belongs in `glci-sv.js`.

Reply:
fixed